docs/interfaces.md

```diff
@@ -0,0 +1,101 @@
+# Interfaces
+
+Interfaces allow snaps to communicate or share resources according to the
+protocol defined by the interface.
```

Comment:
s/defined by/established for/.. subtle, but makes it slightly more clear that this is a convention.